### PR TITLE
Update Spring releases July 2021

### PR DESCRIPTION
**Pull request by saragluna**
For Spring July release:

- Depend on Spring Boot 2.5.2.
- Depend on Spring Cloud 2020.0.3.
- Depend on Spring Security 5.5.1.
- Depend on Spring Integration 5.5.1.
- Depend on Azure Core 1.18.0.
- Depend on Azure Identity 1.3.3
- Depend on Azure Cosmos 4.17.0.
- Depend on Azure Spring Data Cosmos 3.9.0.
- Depend on Azure Key Vault Secret 4.3.1.
- Depend on Azure EventHubs 5.9.0.
- Depend on Azure Service Bus 7.3.0.
- Depend on Azure Storage Blob 12.12.0.
- Depend on Azure Storage File Share 12.10.0.

### CODE REVIEW

File: sdk/spring/azure-spring-boot-starter/CHANGELOG.md [hunk 1]
```diff
@@ -1,6 +1,8 @@
 # Release History
 
-## 3.7.0-beta.1 (Unreleased)
+## 3.7.0 (2021-07-20)
+No changes from 3.6.1.
```
Comment (yiliuTo) [inline sdk/spring/azure-spring-boot-starter/CHANGELOG.md:4]: Should this be align with azure-spring-boot that cover all changes of spring boot starters？

File: sdk/spring/azure-spring-cloud-autoconfigure/CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +11,10 @@
 ### Breaking Changes
 - Encode message payload with UTF-8 charset instead of default charset of JVM when the payload is String. ([#23056](https://github.com/Azure/azure-sdk-for-java/pull/23056))
```
Comment (yiliuTo) [inline sdk/spring/azure-spring-cloud-autoconfigure/CHANGELOG.md:13]: Ops, there is a piece of changelog for servicebus ignored here, could you help to add below content ?
### New Features
- Support configuration of `AmqpTransportType` for ServiceBusClientBuilder with property of `spring.cloud.azure.servicebus.transportType`, supported values are `AMQP` and `AMQP_WEB_SOCKETS`.

**Review by yiliuTo — APPROVED** [review]
LGTM

File: sdk/spring/azure-spring-boot-starter/CHANGELOG.md (hunk 1 above)
Comment (saragluna) [reply sdk/spring/azure-spring-boot-starter/CHANGELOG.md]: changed.

**Review by mrm9084 — APPROVED** [review]

**Comment by saragluna** [comment]
The IT error is transient, so override the check-enforcer.

**Pull request merged**